### DISCUSSION TITLE
Is there a way to force a Tasmota FW upgrade from the hot-spot-web-page?

### DISCUSSION THREAD

**Discussion by st2000** [arendst/Tasmota#10913] [Q&A] [answered]
I can not get my smart switch to connect to my WIFI network.  But I can dependably revert to the hot-spot-web-page using the 6 power cycle method.  I have tried several time so am doubtful I have entered the credentials incorrectly every time.  And I have previously been successful loading Tasmota on another smart switch of the same make / model and connecting that to my WIFI network.

I suspect the current copy of Tasmota inside this smart switch to be corrupted.

So, is there a way to force a Tasmota FW upgrade from the hot-spot-web-page?

**Comment by barbudor** [comment]
When in AP mode (WifiManager) I can see the exact same menus including the "Firmware Upgrade" menu.
Don't you have it ?

**Comment by ascillato** [comment, marked as answer]
When you are in AP mode, you need to go to the page [http://192.168.4.1/up?](http://192.168.4.1/up?)
